starter-code/ShoppingListWithDetailView/app/src/main/java/ly/generalassemb/drewmahrt/shoppinglistdetailview/ShoppingListAdapter.java

```diff
@@ -15,38 +16,28 @@
 
 public class ShoppingListAdapter extends RecyclerView.Adapter<ShoppingItemViewHolder> {
     private List<ShoppingItem> mShoppingItems;
+    ShoppingListFragment.OnShoppingItemListener mListener;
 
-    public ShoppingListAdapter(List<ShoppingItem> shoppingItems) {
+    public ShoppingListAdapter(List<ShoppingItem> shoppingItems, ShoppingListFragment.OnShoppingItemListener listener) {
         mShoppingItems = shoppingItems;
+        mListener = listener;
     }
 
     @Override
     public ShoppingItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
-        return new ShoppingItemViewHolder(LayoutInflater.from(parent.getContext()).inflate(android.R.layout.simple_list_item_1,parent,false));
+        return new ShoppingItemViewHolder(LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.shopping_item,parent,false));
     }
 
     @Override
-    public void onBindViewHolder(final ShoppingItemViewHolder holder, int position) {
+    public void onBindViewHolder(final ShoppingItemViewHolder holder, final int position) {
+        ShoppingItem item = mShoppingItems.get(position);
+        holder.mNameTextView.setText(item.getName());
 
-        final ShoppingItem currentItem = mShoppingItems.get(position);
-
-        holder.mNameTextView.setText(currentItem.getName());
-
-        // Add an OnClickListener that launches DetailActivity and passes it the item's ID
-        holder.mNameTextView.setOnClickListener(new View.OnClickListener() {
+        holder.mActivityDetail.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                // Get a reference to the MainActivity as a Context
-                Context mainActivity = holder.mNameTextView.getContext();
-
-                // Create the intent
-                Intent intent = new Intent(mainActivity, DetailActivity.class);
-
-                // Add the ID as an extra
-                intent.putExtra(DetailActivity.ITEM_ID_KEY, currentItem.getId());
-
-                // Start the detail activity
-                mainActivity.startActivity(intent);
+                mListener.onItemSelected(mShoppingItems.get(holder.getAdapterPosition()));
```

Comment:
If you go the route of passing only the item's ID value from MainActivity to the detail fragment, then here you could send just the item ID (rather than it's position in the adapter) from the adapter back to MainActivity